pkg/controller/clone-controller.go

```diff
@@ -264,10 +296,16 @@ func (r *CloneReconciler) findCloneSourcePod(pvc *corev1.PersistentVolumeClaim)
 	if !isCloneRequest {
 		return nil, nil
 	}
+	cloneSourcePodName, exists := pvc.Annotations[AnnCloneSourcePod]
+	if !exists {
+		// fallback to legacy name, to find any pod that still might be running after upgrade
+		cloneSourcePodName = createCloneSourcePodName(pvc)
+	}
 
+	// TODO: I think we might even get pod by name directly not using label selector
```

Comment:
yeah no need for the label

Reply:
I could have sworn this was something that was changed when this controller was refactored

Reply:
Want to confirm that before this pr, the value of this annotation was equal to the pod name?

Reply:
Interesting, Before this PR the pod name and CloneUniqueID was already the same:
commit: 17db2920b1582979625e9decf6aed572d436bec0
podName := getCloneSourcePodName(targetPvc)				
CloneUniqueID:          getCloneSourcePodName(targetPvc),

but before 17db2920b1582979625e9decf6aed572d436bec0  it was:
	podName := fmt.Sprintf("%s-%s-", common.ClonerSourcePodName, sourcePvcName)
	id := string(pvc.GetUID())
	CloneUniqueID:          id + "-source-pod",

Reply:
so it looks like some steps to change this were taken (but not finished)